### ISSUE TITLE
NVDA in IE 11 reads elements with display: none; and aria-hidden="true "applied within containers with role="dialog" and role="document" after setting focus on visible element

### ISSUE THREAD

**Issue by NataliePakki** [nvaccess/nvda#7585]
NVDA in IE 11 reads elements with display: none; and aria-hidden="true "applied within containers with role="dialog" and role="document" after setting focus on visible element.

In Firefox & Chrome it doesn't read the hidden content, only in IE this is happening.

Example:
- Open IE 11
- Browse to https://jsfiddle.net/xwr27ny2/
- press "Run"

### Expected behavior:
- NVDA doesn't read invisible elements with `aria-hidden="true"` and/or `display:none`
- it must read something like this:
> dialog
> Visible Link  document
> Visible Link  link  Visible Link

### Actual behavior:
- NVDA reads invisible elements with `aria-hidden="true"` and/or `display:none`
- it reads:
> dialog
> **HIDE DIV**
> **HIDE LINK** Visible Link  document
> Visible Link  link  Visible Link

### System configuration:
NVDA version:
- 2017.3

Windows version:
 - 10.0.15063

Name and version of other software in use when reproducing the issue:
- IE 11.540.15063.0

### Other questions:

Does the issue still occur after restarting your PC?
- Yes

Have you tried any other versions of NVDA?
- Not yet.

**Comment by josephsl** [comment]
Hi, please try with NVDA 2017.1 or 2017.2 and see if this persists, as well as try with Edge to find out if it really is an IE/MSHTML thing. Thanks. CC @DerekRiemer

**Comment by NataliePakki** [comment]
@josephsl,

NVDA 2017.3/2017.1 in Edge reads :

> Visible Link
> link
> dialog

NVDA 2017.1/2016.4 in IE11 reads:

> dialog
> **HIDE DIV**
> **HIDE LINK** Visible Link  document
> Visible Link  link  Visible Link
Thanks,

**Comment by Adriani90** [comment]
@NataliePakki It seems JSFiddle is not well supported on IE. Could oyou please post here the code itself? This would make it easy for everyone to reproduce this issue. Thanks.

**Comment by malcalevak** [comment]
Since I just encountered the same issue, here's their code, simplified to use inline CSS so you can copy/paste it:
```
<div role="dialog">
	<div role="document">
      <div aria-hidden="true" style="display: none !important; visibility: hidden;">HIDE DIV</div>
       <a href="javascript:void(0);" aria-hidden="true" hidden>HIDE LINK</a>       
       <a href="javascript:void(0);" id="focus">Visible Link</a>				
  </div>
</div>
```

**Comment by Adriani90** [comment]
This is not in conformity with the Aria spec and NVDA definitely does not react correctly here.
cc: @lukaszgo1 any suggestions what could cause this issue in NVDA?
cc: @feerrenrut

**Comment by Adriani90** [comment]
IE support ended completely in 2022, and the browser is automatically removed from machines that still have it after an Edge update in February 2023. I am closing this as abandoned, since I don't expect any further accessibility updates to IE or to IE mode in Edge. In Edge Chromium 111 I cannot reproduce this issue, Chrome and firefox work also well.
If you have still this issue in other browsers, please open a new issue with updated steps to reproduce and a minimal test case.
